lib/dialects/postgres/query.js

```diff
@@ -131,6 +132,10 @@ Query.prototype.run = function(sql, parameters) {
       }
     }
 
+    if (self.options.ensureAffectedRows && result.rowCount < 1) {
+      throw new sequelizeErrors.NoAffectedRowsError('No rows affected by: ' + sql);
+    }
```

Comment:
It should be possible to handle it by looking at type and return value in `sequelize.query`, if you're seeing different results from the dialects they would need to be fixed first.